### ISSUE TITLE
Unable to reproduce the results

### ISSUE THREAD

**Issue by Lihua101** [PRBonn/LiDiff#41]
Hello Author,

I tried training the diffusion model part using 4*p40_bs=1 to correspond to 2*A6000_bs=2 as described. However, my model only achieves results comparable to those in the paper for IoU (0.5m), while the results for IoU (0.2m) and IoU (0.1m) are lower than those reported in the paper. The details are as follows.

Apart from the batch size (bs), I did not adjust any other parameters. Do you think I need to modify the learning rate as well?

Additionally, I have only trained the diffusion module and have not yet trained the refinement module.

Looking forward to your advice. Thank you！

![Image](https://github.com/user-attachments/assets/6b640df8-e80d-45ac-a060-68216bb8c9e6)
![Image](https://github.com/user-attachments/assets/f1591a5d-4b29-4ee6-bb0a-adf07cf9302a)

**Comment by nuneslu** [comment]
Hi! Did you run the evaluation for how many point clouds? For the whole validation set from SemanticKITTI?

**Comment by Lihua101** [comment]
> Hi! Did you run the evaluation for how many point clouds? For the whole validation set from SemanticKITTI?

During validation, I only sampled 200 random point clouds from the "08" directory, but all results were similar to those mentioned earlier—except for IoU (0.5m), the other metrics (IoU 0.2m and 0.1m) fell far short of the reported values in the paper.

Do you think this could be due to hardware differences or the batch size (bs)? Why is the performance gap so significant?

**Comment by nuneslu** [comment]
Hi, again. Could you share some visualizations from the results? I would also suggest visualizing together the ground truth and the completed scene to see if there is any offset between the point clouds.

Also, you can rerun the evaluation with more than 200 point clouds, this may also be the reason for the different performance than the on in the paper.

**Comment by Lihua101** [comment]
Hi, I attempted to create a visualization, and here is the ground truth along with the weights you provided and the results I obtained using a 4p40_bs=1_epoch=20_lr=0.0001 setup. It seems like the point cloud hasn't converged. Even after training for 10 more epochs, the results still haven't improved. However, since the loss remains around 0.9, I'm having difficulty evaluating the training performance. Do you think I should adjust the learning rate, or do you have any suggestions that could help me replicate the results from the paper?

Thank you for your guidance!

![Image](https://github.com/user-attachments/assets/075318c0-e374-4005-8e78-c5ca35f18585)
![Image](https://github.com/user-attachments/assets/35107e39-712c-459d-9755-ff7760c46542)
![Image](https://github.com/user-attachments/assets/f782f9e8-8c18-4a5c-8df3-61b73245833c)

**Comment by nuneslu** [comment]
Hi, I guess that the problem is not related to the learning rate, but rather to the batch size. My suggestion is to keep bsize=2 and in this [line](https://github.com/PRBonn/LiDiff/blob/b0f2cc0b8f70332ce9ccf255bd35312936f3b944/lidiff/models/models.py#L203) random sample a subsets of points from `denoise_t` and `noise`, and compute the loss only over the subset. This would keep the batch size 2 and still reduce the memory usage.

**Comment by nuneslu** [comment]
Regarding the training, training for more 10 epochs may not help. We also trained for more 10 epochs and noticed that with 20 epochs the results were way better. You can check the results with 15 epochs only.

In our experiments, the loss also converges to 0.9. The diffusion loss doesn't give much insight into the generation quality.

![Image](https://github.com/user-attachments/assets/7a36f81b-5366-4983-bb66-23429e6384df)
